### PR TITLE
urldata: remove values that are not used depending on compilation settings

### PR DESCRIPTION
**Pull request by MAntoniak**
Reduced size of dynamically_allocated_data structure.

Reduced number of stored values in enum dupstring and enum dupblob. This affects the reduced array placed in the UserDefined structure.

It is also possible to remove the STRING_SASL_AUTHZID and STRING_BEARER values from the dupstring. However, this requires a separate PR.

### CODE REVIEW

File: lib/urldata.h [hunk 1]
```diff
@@ -1417,85 +1423,123 @@ struct Curl_multi;    /* declared in multihandle.c */
...
+  STRING_KEY_TYPE,        /* format for private key (default: PEM) */
+  STRING_SSL_CAPATH,      /* CA directory name (doesn't work on windows) */
+  STRING_SSL_CAFILE,      /* certificate file to verify peer against */
+  STRING_SSL_PINNEDPUBLICKEY, /* public key file to verify peer against */
+  STRING_SSL_CIPHER_LIST, /* list of ciphers to use */
+  STRING_SSL_CIPHER13_LIST, /* list of TLS 1.3 ciphers to use */
+  STRING_SSL_CRLFILE,     /* crl file to check certificate */
+  STRING_SSL_ISSUERCERT, /* issuer cert file to check certificate */
+  STRING_SERVICE_NAME,    /* Service name */
+#ifndef CURL_DISABLE_PROXY
```
Comment (bagder) [inline lib/urldata.h:1438]: This is in direct contradiction to the comment above the enum which says:

**This enumeration MUST not use conditional directives (#ifdefs)**

But: the comment also refers to a (misspelled) source code file that was removed in 821d108fc9adfb, which makes me suspect maybe this comment does not apply anymore?

@monnerat can you fill in some blanks maybe?

Comment (monnerat) [reply lib/urldata.h:1438]: It seems the comment is obsolete: the current test is now based on `CURLOPT_*`, not on `STRING_*` anymore.

Comment (bagder) [reply lib/urldata.h:1438]: @MAntoniak can you perhaps update that comment to reflect reality?

Comment (MAntoniak) [reply lib/urldata.h:1438]: If I understand correctly just remove it?

Comment (bagder) [reply lib/urldata.h:1430]: Yes!

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**